File: python solutions/binary_tree_postorder_traversal.py
# Definition for a binary tree node.
from typing import List, Optional


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def postorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        def dfs(node):
            if not node: return

            dfs(node.left)
            dfs(node.right)
            
            ans.append(node.val)

            return
        ans = []
        dfs(root)
        return ans

File: ruby solutions/binary_tree_postorder_traversal.rb
# Definition for a binary tree node.
class TreeNode
    attr_accessor :val, :left, :right
    def initialize(val = 0, left = nil, right = nil)
        @val = val
        @left = left
        @right = right
    end
end
# @param {TreeNode} root
# @return {Integer[]}
def postorder_traversal(root)
    def dfs(node)
        return if !node

        dfs(node.left)
        dfs(node.right)

        @ans.append(node.val)
        return
    end
    @ans = []
    dfs(root)
    return @ans
end